test/battle/ability/clear_body.c

```diff
@@ -26,13 +26,211 @@ SINGLE_BATTLE_TEST("Clear Body prevents intimidate")
-TO_DO_BATTLE_TEST("Clear Body prevents Sticky Web");
-TO_DO_BATTLE_TEST("Clear Body doesn't prevent stat stage reduction from moves used by the user"); // e.g. Superpower
-TO_DO_BATTLE_TEST("Clear Body doesn't prevent Speed reduction from Iron Ball");
-TO_DO_BATTLE_TEST("Clear Body doesn't prevent Speed reduction from paralysis");
-TO_DO_BATTLE_TEST("Clear Body doesn't prevent Attack reduction from burn");
-TO_DO_BATTLE_TEST("Clear Body doesn't prevent receiving negative stat changes from Baton Pass");
-TO_DO_BATTLE_TEST("Clear Body doesn't prevent Topsy-Turvy");
-TO_DO_BATTLE_TEST("Clear Body doesn't prevent Spectral Thief from resetting positive stat changes");
-TO_DO_BATTLE_TEST("Clear Body is ignored by Mold Breaker");
+SINGLE_BATTLE_TEST("Clear Body prevents stat stage reduction from moves")
+{
+    u16 move;
+    PARAMETRIZE{ move = MOVE_GROWL; }
+    PARAMETRIZE{ move = MOVE_LEER; }
+    PARAMETRIZE{ move = MOVE_CONFIDE; }
+    PARAMETRIZE{ move = MOVE_FAKE_TEARS; }
+    PARAMETRIZE{ move = MOVE_SCARY_FACE; }
+    PARAMETRIZE{ move = MOVE_SWEET_SCENT; }
+    PARAMETRIZE{ move = MOVE_SAND_ATTACK; }
+
+    GIVEN {
+        ASSUME(gMovesInfo[MOVE_GROWL].effect == EFFECT_ATTACK_DOWN);
+        ASSUME(gMovesInfo[MOVE_LEER].effect == EFFECT_DEFENSE_DOWN);
+        ASSUME(gMovesInfo[MOVE_CONFIDE].effect == EFFECT_SPECIAL_ATTACK_DOWN);
+        ASSUME(gMovesInfo[MOVE_FAKE_TEARS].effect == EFFECT_SPECIAL_DEFENSE_DOWN);
+        ASSUME(gMovesInfo[MOVE_SCARY_FACE].effect == EFFECT_SPEED_DOWN_2);
+        ASSUME(gMovesInfo[MOVE_SWEET_SCENT].effect == EFFECT_EVASION_DOWN);
+        ASSUME(gMovesInfo[MOVE_SAND_ATTACK].effect == EFFECT_ACCURACY_DOWN);
+        PLAYER(SPECIES_WOBBUFFET) { Moves(move); }
+        OPPONENT(SPECIES_BELDUM) { Ability(ABILITY_CLEAR_BODY); }
+    } WHEN {
+        TURN{ MOVE(player, move); }
+    } SCENE {
+        NONE_OF { ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_STATS_CHANGE, opponent); }
+        ABILITY_POPUP(opponent, ABILITY_CLEAR_BODY);
+        MESSAGE("Foe Beldum's Clear Body prevents stat loss!");
+    }
+}
+
+SINGLE_BATTLE_TEST("Clear Body prevents Sticky Web")
+{
+    GIVEN {
+        ASSUME(gMovesInfo[MOVE_STICKY_WEB].effect == EFFECT_STICKY_WEB);
+        PLAYER(SPECIES_WOBBUFFET)
+        OPPONENT(SPECIES_WOBBUFFET)
+        OPPONENT(SPECIES_BELDUM) { Ability(ABILITY_CLEAR_BODY); }
+    } WHEN {
+        TURN{ MOVE(player, MOVE_STICKY_WEB); MOVE(opponent, MOVE_CELEBRATE); }
+        TURN{ SWITCH(opponent, 1); }
+    } SCENE {
+        NONE_OF { ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_STATS_CHANGE, opponent); }
+        ABILITY_POPUP(opponent, ABILITY_CLEAR_BODY);
+        MESSAGE("Foe Beldum's Clear Body prevents stat loss!");
+    }
+}
+
+SINGLE_BATTLE_TEST("Clear Body doesn't prevent stat stage reduction from moves used by the user")
+{
+    GIVEN {
+        ASSUME(gMovesInfo[MOVE_SUPERPOWER].additionalEffects->moveEffect == MOVE_EFFECT_ATK_DEF_DOWN);
```

Comment:
There's a function called HasMoveEffect or something like that should be used in case, as move effects can be in other positions besides the first one.